src/main/java/net/dv8tion/jda/api/sharding/DefaultShardManagerBuilder.java

```diff
@@ -436,7 +478,23 @@ public static DefaultShardManagerBuilder create(@Nullable String token, @Nonnull
     @CheckReturnValue
     public static DefaultShardManagerBuilder create(@Nullable String token, @Nonnull Collection<GatewayIntent> intents)
     {
-        return new DefaultShardManagerBuilder(token, GatewayIntent.getRaw(intents));
+        return new DefaultShardManagerBuilder(token, GatewayIntent.getRaw(intents)).applyIntents();
+    }
+
+    private DefaultShardManagerBuilder applyIntents()
+    {
+        EnumSet<CacheFlag> disabledCache = EnumSet.allOf(CacheFlag.class);
+        for (CacheFlag flag : CacheFlag.values())
+        {
+            GatewayIntent requiredIntent = flag.getRequiredIntent();
+            if (requiredIntent == null || (requiredIntent.getRawValue() & intents) != 0)
+                disabledCache.remove(flag);
+        }
```

Comment:
I really do think we need a warning here letting users know that the flag was automatically marked disabled. It concerns me that users won't understand why certain functions don't work if they don't know what cache flags are when they use those functions.

A function might say it doesn't work if X cache flag is disabled, but if a new person doesn't know what a cache flag is but _does_ know that they didn't disable any of them, then they'll be confused